src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerMiddleware.cs

```diff
@@ -22,13 +27,44 @@ public class IISPlatformHandlerMiddleware
         private const string XOriginalIPName = "X-Original-IP";
 
         private readonly RequestDelegate _next;
+        private readonly IISPlatformHandlerOptions _options;
 
-        public IISPlatformHandlerMiddleware(RequestDelegate next)
+        public IISPlatformHandlerMiddleware(RequestDelegate next, IISPlatformHandlerOptions options)
         {
+            if (next == null)
+            {
+                throw new ArgumentNullException(nameof(next));
+            }
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
             _next = next;
+            _options = options;
         }
 
-        public Task Invoke(HttpContext httpContext)
+        public async Task Invoke(HttpContext httpContext)
+        {
+            CheckScheme(httpContext);
```

Comment:
Why Check? UpdateScheme, UpdateRemoteIp? TryUpdate. This operation potentially mutates the HttpContext so Check is a misnomer